File: test-projects/remote-deps/ScalaTest/src/test/scala/StatisticsTest.scala
import minitest._
import synapses.Library._

object StatisticsTest extends SimpleTestSuite {

  val expectedWithOutputValues =
    LazyList(
      (List(0.0, 0.0, 1.0), List(0.0, 0.0, 1.0)),
      (List(0.0, 0.0, 1.0), List(0.0, 1.0, 1.0))
    )

  test("root mean square error") {
    assertEquals(
      Statistics.rootMeanSquareError(expectedWithOutputValues),
      0.7071067811865476
    )
  }

}

File: test-projects/remote-deps/ElixirTest/test/statistics_test.exs
defmodule StatisticsTest do
  use ExUnit.Case
  alias Synapses.{Statistics}

  def expectedWithOutputValues(),
      do: [
            {[0.0, 0.0, 1.0], [0.0, 0.0, 1.0]},
            {[0.0, 0.0, 1.0], [0.0, 1.0, 1.0]}
          ]
          |> Stream.map(fn x -> x end)

  test "root mean square error" do
    assert expectedWithOutputValues() |> Statistics.rootMeanSquareError == 0.7071067811865476
  end
end
